UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/AdmissionContract.java

```diff
@@ -0,0 +1,136 @@
+package de.upb.cs.uc4.chaincode;
+
+import de.upb.cs.uc4.chaincode.exceptions.LedgerAccessError;
+import de.upb.cs.uc4.chaincode.model.*;
+import de.upb.cs.uc4.chaincode.model.errors.InvalidParameter;
+import de.upb.cs.uc4.chaincode.util.AdmissionContractUtil;
+import de.upb.cs.uc4.chaincode.util.GsonWrapper;
+import org.hyperledger.fabric.contract.Context;
+import org.hyperledger.fabric.contract.annotation.Contract;
+import org.hyperledger.fabric.contract.annotation.Default;
+import org.hyperledger.fabric.contract.annotation.Transaction;
+import org.hyperledger.fabric.shim.ChaincodeStub;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Contract(
+        name="UC4.Admission"
+)
+@Default
+public class AdmissionContract extends ContractBase {
+    private final AdmissionContractUtil cUtil = new AdmissionContractUtil();
+
+    protected String contractName = "UC4.Admission";
+
+    /**
+     * Adds MatriculationData to the ledger.
+     * @param ctx transaction context providing access to ChaincodeStub etc.
+     * @param admissionJson json representation of new Admission to add.
+     * @return newAdmission on success, serialized error on failure
+     */
+    @Transaction()
+    public String addAdmission(final Context ctx, String admissionJson) {
+        ChaincodeStub stub = ctx.getStub();
+
+        Admission newAdmission;
+        try {
+            newAdmission = GsonWrapper.fromJson(admissionJson, Admission.class);
+            newAdmission.resetAdmissionId();
+        } catch (Exception e) {
+            return GsonWrapper.toJson(cUtil.getUnprocessableEntityError(cUtil.getUnparsableParam("admission")));
+        }
+
+        if (cUtil.keyExists(stub, newAdmission.getAdmissionId())) {
+            return GsonWrapper.toJson(cUtil.getConflictError());
+        }
+
+        ArrayList<InvalidParameter> invalidParams = cUtil.getParameterErrorsForAdmission(newAdmission);
+        if (!invalidParams.isEmpty()) {
+            return GsonWrapper.toJson(cUtil.getUnprocessableEntityError(invalidParams));
+        }
+
+        if (cUtil.keyExists(stub, newAdmission.getAdmissionId())) {
+            return GsonWrapper.toJson(cUtil.getConflictError());
+        }
```

Comment:
Is there a reason to check for conflict twice?

Reply:
```suggestion
```